var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -337,11 +337,14 @@ def setup_build_environment(self, env):
         # https://bugzilla.redhat.com/show_bug.cgi?id=1795817
         if self.spec.satisfies('%gcc@10:'):
             env.set('FFLAGS', '-fallow-argument-mismatch')
+            env.set('FCFLAGS', '-fallow-argument-mismatch')
         # Same fix but for macOS - avoids issue #17934
         if self.spec.satisfies('%apple-clang@11:'):
             env.set('FFLAGS', '-fallow-argument-mismatch')
+            env.set('FCFLAGS', '-fallow-argument-mismatch')
```

Comment:
Should this not query the Fortran compiler instead? The `apple-clang`/`clang` version is not inherently related to the gfortran version.

Reply:
Not sure exactly how to do that though.

Reply:
The unspoken rule in spack is that `clang` goes with Fortran, unfortunately ... this code is not from me. But I can try to fix it.

Reply:
I've done that before (see esmf `package.py`)